FirebaseAuth.podspec

```diff
@@ -40,16 +40,37 @@ supports email and password accounts, as well as several 3rd party authenticatio
     source + '**/*.[mh]',
     'FirebaseCore/Extension/*.h',
     'FirebaseAuth/Interop/*.h',
+    source + 'FirebaseAuth.private.modulemap'
   ]
+  s.module_map = source + 'Public/FirebaseAuth/FirebaseAuth.modulemap'
   s.public_header_files = source + 'Public/FirebaseAuth/*.h'
+
+  # TODO(ncooke3): Maybe set this to a filtering of `s.source_files`
+  s.private_header_files = [
+    # All headers except the ones in the `Public` should be private.
+    source + 'Private/**/*.h',
+    source + 'Auth/**/*.h',
+    source + 'Backend/**/*.h',
+    source + 'MultiFactor/**/*.h',
+    source + 'Swift/**/*.h',
+    source + 'SystemService/**/*.h',
+    source + 'User/**/*.h',
+    source + 'Utilities/**/*.h',
+    'FirebaseCore/Extension/*.h',
+    'FirebaseAuth/Interop/*.h'
+  ]
+
   s.preserve_paths = [
     'FirebaseAuth/README.md',
     'FirebaseAuth/CHANGELOG.md'
   ]
   s.pod_target_xcconfig = {
     'GCC_C_LANGUAGE_STANDARD' => 'c99',
     # The second path is to find FirebaseAuth-Swift.h from a pod gen project
-    'HEADER_SEARCH_PATHS' => '"${PODS_TARGET_SRCROOT}" "${OBJECT_FILE_DIR_normal}/${NATIVE_ARCH_ACTUAL}"'
+    'HEADER_SEARCH_PATHS' => '"${PODS_TARGET_SRCROOT}" "${OBJECT_FILE_DIR_normal}/${NATIVE_ARCH_ACTUAL}"',
+    # Point to the private module map.
+    'MODULEMAP_PRIVATE_FILE' =>
```

Comment:
What does CocoaPods with module maps and private headers if there is not an explicit one supplied?